64jam.p8

```diff
@@ -47,57 +90,19 @@ m = false --to stop double move
 		started = true
 	end
 
-if btn (1) and btn (2) then 
-		player.dir = 1
-		player.x += 1
-		player.y -= 1
-		m = true
-	end
-	if btn (1) and btn (3) then 
-		player.dir = 5
-		player.x += 1
-		player.y += 1
-		m = true 
-	end
-	if btn (0) and btn (2) then 
-		player.dir = 10
-		player.x -= 1
-		player.y -= 1
-		m = true 
-	end
-	if btn (0) and btn (3) then 
-		player.dir = 7
-		player.x -= 1
-		player.y += 1 
-		m = true
-	end
-	if (m== false) then
-		if btn(0) then 
-			player.dir = 9
-			player.x -= 1 
-		end
-		if btn(1) then 
-			player.dir = 3 
-			player.x += 1
-		end
-		if btn(2) then 
-			player.dir = 12
-			player.y -= 1
-	 end
-		if btn(3) then 
-			player.dir = 6
-			player.y += 1 
-		end
-	end
-	
+	set_player_dir_from_buttons()
+
+	update_player()
 	update_projectiles()
-	if started then player.fuel -= 1 end
 	
 	if btn(5) then fire_bullet() end
- if btnp(4) then fire_rocket() end
+	if btnp(4) then fire_rocket() end
 end
 
 function update_projectiles()
+	-- todo, combine projectiles arrays
+	-- todo, track projectile fuel, instead of using distance from player
```

Comment:
Interesting idea.  I initially used the distance from player to remove bullets that had gone past the bounds of the screen.  I wanted to nerf the rockets a little by making them slightly more close range (so the player can't cheese the game by hammering things just off screen).